cli/command/cli.go

```diff
@@ -324,7 +324,7 @@ func (cli *DockerCli) getInitTimeout() time.Duration {
 
 func (cli *DockerCli) initializeFromClient() {
 	ctx := context.Background()
-	if !strings.HasPrefix(cli.DockerEndpoint().Host, "ssh://") {
+	if !strings.HasPrefix(cli.client.DaemonHost(), "ssh://") {
```

Comment:
Hm... looks like something to fix; perhaps `DaemonHost()` does not contain the `ssh://` protocol, in which case it ignores this block 🤔 

```
#5 [1/4] FROM docker.io/library/docker:test-dind@sha256:af3033aaad3b67cfd2fcefed035ea60b70627d3e7636987ba03f7b3f09f3fc3b
Pulling registry (registry:2)...
2: Pulling from library/registry
Digest: sha256:2e830e8b682d73a1b70cac4343a6a541a87d5271617841d87eeb67a824a5b3f2
Status: Downloaded newer image for registry:2
Creating cliendtoendsuite_evil-notary-server_1 ... 
Creating cliendtoendsuite_notary-server_1      ... 
Creating cliendtoendsuite_engine_1             ... 
Creating cliendtoendsuite_registry_1           ... 
Creating cliendtoendsuite_engine_1             ... done
Creating cliendtoendsuite_evil-notary-server_1 ... done
Creating cliendtoendsuite_registry_1           ... done
Creating cliendtoendsuite_notary-server_1      ... done
Waiting for docker daemon to become available at ssh://penguin@172.18.0.3
time="2022-11-04T15:46:30Z" level=warning msg="commandConn.CloseWrite: commandconn: failed to wait: signal: killed"
time="2022-11-04T15:46:30Z" level=warning msg="commandConn.CloseWrite: commandconn: failed to wait: signal: killed"
Client:
 Version:           22.06.0-dev
 API version:       1.40 (downgraded from 1.42)
 Go version:        go1.19.2
 Git commit:        d9474bc
 Built:             Fri Nov  4 15:43:50 2022
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46
  Built:            Wed Sep 16 17:04:43 2020
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
time="2022-11-04T15:46:30Z" level=warning msg="commandConn.CloseWrite: commandconn: failed to wait: signal: killed"
error during connect: Post "http://docker.example.com/v1.40/images/create?fromImage=alpine&tag=sha256%3Af006ecbb824d87947d0b51ab8488634bf69fe4094959d935c0c103f4820a417d": command [ssh -l penguin -- 172.18.0.3 docker system dial-stdio] has exited with signal: killed, please make sure the URL is valid, and Docker 18.09 or later is installed on the remote host: stderr=
make: *** [docker.Makefile:145: test-e2e-connhelper-ssh] Error 1
Error: Process completed with exit code 2.
```

Reply:
Hm... yes, looks like it. We set up a dialer, but configure the client itself to use "a host" (`http://docker.example.com`);
https://github.com/docker/cli/blob/759007c8ca0cbd6c38e74ec32d75a9720eedd9b2/cli/connhelper/connhelper.go#L43-L54